### PR TITLE
📊 demography: sex gap contribution by age, modal age

### PR DESCRIPTION
**Pull request by lucasrodes**
## Add two new indicators

- [Death rates at specific life stages mold the sex gap in life expectancy](http://staging-site-lucas/admin/datasets/6289)
  - As part of https://github.com/owid/owid-issues/issues/1222
- [Modal age](http://staging-site-lucas/admin/datasets/6290)
  - As part of https://github.com/owid/owid-issues/issues/1216

## Other improvements
- `owid.catalog.processing.read_rda`: Read `rda` files.
- `etl.snapshot.Snapshot.extract`: Extract a snapshot zip file to a folder.
- Add license `MIT License` in wizard options.

### CODE REVIEW

**Comment by spoonerf** [comment]
Hey @lucasrodes - have you uploaded the snapshot file of `modal_age_death`, I am getting this error when running the snapshot step: 

`Exception: File not found in DVC. Have you run the snapshot script? Make sure you're not using is_public: false`

**Comment by lucasrodes** [comment]
Hey, @spoonerf! I think something occurred overnight that deleted these. I just uploaded them again, sorry!

File: etl/steps/data/garden/demography/2023-11-08/modal_age_death.meta.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    # http://docs.owid.io/projects/etl/architecture/metadata/reference/indicator/
+    variables:
+      modal_age_death:
+        title: Adult modal age at death
+        unit: years
+        description_short: |-
+          The most common age at death, among adults.
+
+        description_key:
+          - The adult modal age at death is the age (beyond infancy) at which the majority of deaths occur.
```
Comment (spoonerf) [inline etl/steps/data/garden/demography/2023-11-08/modal_age_death.meta.yml:29]: Perhaps update this in line with the `The most common age at which people die in a given year.`, based on the feedback [here](https://owid.slack.com/archives/C04LNF8LBA8/p1699525970163009). Also maybe clarify exactly what infant means here? I think it is typically used to mean < 1 year old, but is it the same here, I see 10 being mentioned later.

File: etl/steps/data/garden/demography/2023-11-08/modal_age_death.meta.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          - In a recent paper by Diaconu et al. (2022), the authors propose the modal age at death for measuring occupational inequalities at older ages in Finland. They compare trends in modal age at death with those of more conventional measures of mortality at older ages, such as life expectancy at age 65, 75.
+          - |-
+            Compared to life expectancy at birth or at age 65, M has the advantages of being: (i) determined solely by mortality at older ages, (ii) free from an arbitrary choice of "old" age threshold, and (iii) compares individuals with similar survival chances over time.
```
Comment (spoonerf) [inline etl/steps/data/garden/demography/2023-11-08/modal_age_death.meta.yml:32]: Maybe worth adding what 'M' means here, I think it's clarified in the dataset description but could be worth adding again.

File: etl/steps/data/garden/demography/2023-11-08/le_sex_gap_age_contribution.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    tb = tb[COLUMNS_KEEP.keys()].rename(columns=COLUMNS_KEEP)
+
+    # Harmonize country names
+    tb = geo.harmonize_countries(df=tb, countries_file=paths.country_mapping_path)
+
+    # Age group
+    tb["age_group"] = tb["age_group"].map(AGE_GROUPS_MAPPING)
+
+    # Get relative value as a percentage
+    tb["ctb_rel"] *= 100
```
Comment (spoonerf) [inline etl/steps/data/garden/demography/2023-11-08/le_sex_gap_age_contribution.py:50]: There are some negative values in the data e.g. France, 15-39 years, 1861, I suppose this makes sense in the context and is expected?

**Review by spoonerf — APPROVED** [review]
Looks good, just a couple of minor comments! There are some negative values in the data which I wasn't expecting, just mentioning in case they are unexpected to you too!

File: etl/steps/data/garden/demography/2023-11-08/modal_age_death.meta.yml (hunk 1 above)
Comment (lucasrodes) [reply etl/steps/data/garden/demography/2023-11-08/modal_age_death.meta.yml]: Thanks for the link. I've improved this section.

**Pull request merged**